lib/data_magic/document_builder.rb

```diff
@@ -84,7 +94,7 @@ def fix_field_type(type, value, key=nil)
                     when "lowercase_name"
                       value.to_s.downcase
                     else # "string"
-                      value.to_s
+                      value == 'NULL' ? nil : value.to_s
```

Comment:
the 'NULL' value can actually be configured, so we don't want to do this here.  The logic is actually elsewhere, but called later... ideally this would be refactored to be in one place.